File: C PLusPlus Versions/Question3/Question3.cpp
// Question3.cpp : Defines the entry point for the console application.
//
/*

	Question3
	C++
	Adam Giaccaglia

*/
#include "stdafx.h"
#include <iostream>
#include <string>

using namespace std;

class Bank
{
	friend class Manager;
	private:
		string name;
	protected:
		string securityInfo;
	public: 
		Bank(){
			securityInfo = "High";
		}
		void display(){
			cout <<("This is a bank!");
		}

};

class Manager
{
	
private:
	int id;
public:
	void display()
	{
		cout <<("I am a manager!");
	}
	void securityAccess(Bank x){
		cout <<("Security Information is: " + x.securityInfo);
	}
};

int _tmain(int argc, _TCHAR* argv[])
{
	Bank bank1;
	Manager boss;
	boss.securityAccess(bank1);

	return 0;
}

File: Java-Versions/Question3/src/Corporation/Question3.java

/*
    Question 3
    Java
    Ada Giaccaglia

 */


package Corporation;
public class Question3 {
    public static void main( String[] args ) {
        Bank bank1 = new Bank();
        Manager boss = new Manager();
        boss.securityAccess(bank1);
    }
}

class Bank
{
    private String name;
    protected String securityInfo;
    //so securityInfo is not null
    Bank(){
        securityInfo = "High";
    }
    public void display(){
        System.out.println("This is a bank!");
    }

}
class Manager
{
    private int id;
    public void display()
    {
        System.out.println("I am a manager!");
    }
    public void securityAccess(Bank x){
        System.out.println("Security Information is: " + x.securityInfo);
    }
}